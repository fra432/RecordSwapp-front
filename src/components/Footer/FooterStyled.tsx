import styled from "styled-components";

const FooterStyled = styled.div`
  margin-top: 1rem;
  width: 100%;
  height: 5rem;
  background-color: #dae1e9;
  display: flex;
  align-items: center;

  .logo {
    width: 7rem;
  }

  span {
    position: absolute;
    left: 50%;
    width: 13rem;
    transform: translate(-6.5rem);
    font-weight: 600;
    font-size: 1.2rem;
    justify-self: center;
  }
`;

export default FooterStyled;
