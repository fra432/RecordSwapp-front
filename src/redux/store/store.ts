import { configureStore } from "@reduxjs/toolkit";
import recordSlice from "../features/recordSlice";
import recordsReducer from "../features/recordsSlice";
import userReducer from "../features/userSlice";
import usersReducer from "../features/usersSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    users: usersReducer,
    records: recordsReducer,
    record: recordSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
