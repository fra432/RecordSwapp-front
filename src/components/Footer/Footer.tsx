import FooterStyled from "./FooterStyled";

const Footer = () => {
  return (
    <FooterStyled>
      <img
        src="/images/RecordSwapp-logo.png"
        alt="RecordSwapp logo"
        className="logo"
      />

      <span>© 2022 RecordSwapp®</span>
    </FooterStyled>
  );
};

export default Footer;
