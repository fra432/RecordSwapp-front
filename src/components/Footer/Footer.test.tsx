import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import reactTestRenderer from "react-test-renderer";
import store from "../../redux/store/store";
import Footer from "./Footer";

describe("Given a Footer component function", () => {
  describe("When invoked", () => {
    test("Then it should render correctly a footer with the app logo and a span with '© 2022 RecordSwapp®<'", () => {
      const footer = reactTestRenderer
        .create(
          <BrowserRouter>
            <Provider store={store}>
              <Footer />
            </Provider>
          </BrowserRouter>
        )
        .toJSON();

      expect(footer).toMatchSnapshot();
    });
  });
});
