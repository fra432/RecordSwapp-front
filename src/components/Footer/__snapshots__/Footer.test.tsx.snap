// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Footer component function When invoked Then it should render correctly a footer with the app logo and a span with '© 2022 RecordSwapp®<' 1`] = `
<div
  className="sc-bczRLJ hPZnSM"
>
  <img
    alt="RecordSwapp logo"
    className="logo"
    src="/images/RecordSwapp-logo.png"
  />
  <span>
    © 2022 RecordSwapp®
  </span>
</div>
`;
